x-pack/test/apm_api_integration/tests/traces/trace_by_id.spec.ts

```diff
@@ -4,84 +4,221 @@
+              .errors(
+                instanceJava
+                  .error('[ResponseError] index_not_found_exception')
+                  .timestamp(timestamp + 50)
+              )
+              .children(
+                instanceJava
+                  .span('get_green_apple_🍏', 'db', 'elasticsearch')
+                  .timestamp(timestamp + 50)
+                  .duration(900)
+                  .success()
+              ),
+          ];
+        });
+      const entities = events.toArray();
+      serviceATraceId = entities.slice(0, 1)[0]['trace.id']!;
 
-    it('returns the correct status code', async () => {
-      expect(response.status).to.be(200);
+      await synthtraceEsClient.index(new EntityArrayIterable(entities));
     });
 
-    it('returns the correct number of buckets', async () => {
-      expectSnapshot(response.body.errorDocs.map((doc) => doc.error?.exception?.[0]?.message))
-        .toMatchInline(`
-        Array [
-          "Test CaptureError",
-          "Uncaught Error: Test Error in dashboard",
-        ]
-      `);
-      expectSnapshot(
-        response.body.traceDocs.map((doc) =>
-          'span' in doc ? `${doc.span.name} (span)` : `${doc.transaction.name} (transaction)`
-        )
-      ).toMatchInline(`
-        Array [
-          "/dashboard (transaction)",
-          "GET /api/stats (transaction)",
-          "APIRestController#topProducts (transaction)",
-          "Parsing the document, executing sync. scripts (span)",
-          "GET /api/products/top (span)",
-          "GET /api/stats (span)",
-          "Requesting and receiving the document (span)",
-          "SELECT FROM customers (span)",
-          "SELECT FROM order_lines (span)",
-          "http://opbeans-frontend:3000/static/css/main.7bd7c5e8.css (span)",
-          "SELECT FROM products (span)",
-          "SELECT FROM orders (span)",
-          "SELECT FROM order_lines (span)",
-          "Making a connection to the server (span)",
-          "Fire \\"load\\" event (span)",
-          "empty query (span)",
-        ]
-      `);
-      expectSnapshot(response.body.exceedsMax).toMatchInline(`false`);
+    after(() => synthtraceEsClient.clean());
+
+    describe('return trace', () => {
+      let traces: Awaited<ReturnType<typeof fetchTraces>>['body'];
+      before(async () => {
+        const response = await fetchTraces({
+          traceId: serviceATraceId,
+          query: { start: new Date(start).toISOString(), end: new Date(end).toISOString() },
+        });
+        expect(response.status).to.eql(200);
+        traces = response.body;
+      });
+      it('returns some errors', () => {
+        expect(traces.errorDocs.length).to.be.greaterThan(0);
+        expect(traces.errorDocs[0].error.exception?.[0].message).to.eql(
+          '[ResponseError] index_not_found_exception'
+        );
+      });
+
+      it('returns some trace docs', () => {
+        expect(traces.traceDocs.length).to.be.greaterThan(0);
```

Comment:
I think we can be a little bit more explicit here, e.g. collect spans/transactions/errors separately and having more explicit assertions, like `expect(item.span.name).to.eql('get_green_apple_🍏')`.